src/audio/channels/noise.rs

```diff
@@ -0,0 +1,176 @@
+use super::*;
+
+const MAX_LEN: u8 = 64;
+
+pub struct Noise {
```

Comment:
Are these files for testing audio? Are they producing square wave and so on?

Reply:
They are for producing the audio. The game boy has 4 audio channels - 2 square waves, one sampler, and one pseudo white noise generator. They each have a struct that contains their state and generates the audio samples from that state.

The "public registers" are the data modified by the CPU. They exist in the main memory space - see audio/mod.rs `impl MemDevice for AudioDevice` - they live on those addresses. The "internal registers" contain internal state for each channel, which determines the frequency, volume, etc of the audio samples produced.